src/GitTools.jl

```diff
@@ -65,22 +65,43 @@ function transfer_progress(progress::Ptr{LibGit2.TransferProgress}, p::Any)
     return Cint(0)
 end
 
+const GIT_REGEX =
+    r"^(?:(?:git|ssh|https)(?:://|@))?(?:[\w\.\+\-:]+@)?([^/:]+)[:/](.+?)(?:\.git)?$"i
+const GITHUB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
+const GITLAB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
+const FALLBACK_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
```

Comment:
Can't we just use a single `GIT_PROTOCOL` here?

Reply:
We could, but people might need to use different protocols for different sites. This at least leaves them the option to do that.